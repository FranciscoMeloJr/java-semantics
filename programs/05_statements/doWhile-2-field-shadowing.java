// In this test doWhile body declares a variable
// with the same name as the field. After doWhile completes,
// we should be able to access the field again.

public class main {
  int i = 1;

	void main(String[] args) {
    do {
      int i = 10;
    } while (false);
    System.out.println("After doWhile i = "+ i);
    System.out.println("Done!");
  }
}

// After doWhile i = 1
// Done!
