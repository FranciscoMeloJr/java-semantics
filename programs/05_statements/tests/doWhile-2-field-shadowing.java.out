After doWhile i = 1
Done!
